### PR TITLE
Document models

### PR DESCRIPTION
**Pull request by dannygoldstein**
This PR documents each class in `baselayer.app.models`. The methods, relationships, properties, and `Column`s of each documented class are also documented.

### CODE REVIEW

**Comment by acrellin** [comment]
@dannygoldstein did you see this? https://travis-ci.org/github/cesium-ml/baselayer/builds/719200884#L780

File: app/models.py [hunk 1]
```diff
@@ -182,85 +231,192 @@ def join_model(
...
-        sa.String, nullable=False, unique=True, default=lambda: str(uuid.uuid4())
+        sa.String,
+        nullable=False,
+        unique=True,
+        default=lambda: str(uuid.uuid4()),
+        doc="The name of the token.",
     )
 
     def is_owned_by(self, user_or_token):
+        """Return a boolean indicating whether a User (or Token) has read access
```
Comment (acrellin) [inline app/models.py:403]: This checks whether a token instance is owned by the specified user (or has the same identity as the specified token, if a token is passed in here)

File: app/models.py [hunk 2]
```diff
@@ -82,10 +117,24 @@ def get_if_owned_by(cls, ident, user, options=[]):
         return obj
 
     def is_owned_by(self, user):
+        """Return a boolean indicating whether a User has read access to this
+        object.
+
+        Parameters
+        ----------
+        user : `baselayer.app.models.User`
```
Comment (acrellin) [inline app/models.py:125]: `user` can also be a `Token` instance, I believe

File: app/models.py [hunk 3]
```diff
@@ -68,12 +82,33 @@ def __repr__(self):
...
     def get_if_owned_by(cls, ident, user, options=[]):
+        """Return an object from the database if the requesting user has access
+        to read the object. If the requesting user does not have access, raise
+        an AccessError.
+
+        Parameters
+        ----------
+        ident : integer or string
+           Primary key of the requested object.
+        user : `baselayer.app.models.User`
```
Comment (acrellin) [inline app/models.py:102]: or `Token`

File: app/models.py (hunk 1 above)
Comment (dannygoldstein) [reply app/models.py:401]: fixed

File: app/models.py (hunk 2 above)
Comment (dannygoldstein) [reply app/models.py:125]: fixed

File: app/models.py (hunk 3 above)
Comment (dannygoldstein) [reply app/models.py]: fixed

**Comment by dannygoldstein** [comment]
@profjsb  need to merge this to merge skyportal/skyportal#729

**Pull request merged**